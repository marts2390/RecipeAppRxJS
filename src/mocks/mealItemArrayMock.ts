const mealItemArrayMock = [
  {
    idMeal: '52874',
    strMeal: 'Beef and Mustard Pie',
    strMealThumb: 'https://www.themealdb.com/images/media/meals/sytuqu1511553755.jpg',
  },
  {
    idMeal: '52878',
    strMeal: 'Beef and Oyster pie',
    strMealThumb: 'https://www.themealdb.com/images/media/meals/wrssvt1511556563.jpg',
  },
  {
    idMeal: '52997',
    strMeal: 'Beef Banh Mi Bowls with Sriracha Mayo, Carrot & Pickled Cucumber',
    strMealThumb: 'https://www.themealdb.com/images/media/meals/z0ageb1583189517.jpg',
  }
]

export default mealItemArrayMock
