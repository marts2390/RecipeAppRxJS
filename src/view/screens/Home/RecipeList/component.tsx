import React, { FC } from 'react'
import { Grid } from '@material-ui/core'

import RecipeCard from 'view/components/RecipeCard'

import { IMealsDataModel } from 'models/mealsDataModel'

interface IHomeComponentProps {
  meals: IMealsDataModel[] | null
}

const HomeComponent:FC<IHomeComponentProps> = ({ meals }) => (
  <div>
    <Grid
      container
      spacing={ 3 }
    >
      {meals?.map((item) => (
        <Grid
          item
          xs={ 6 }
          sm={ 4 }
          md={ 3 }
          lg={ 2 }
          xl={ 1 }
          key={ item.idMeal }
        >
          <RecipeCard item={ item } />
        </Grid>
      ))}
    </Grid>
  </div>
)

export default HomeComponent
