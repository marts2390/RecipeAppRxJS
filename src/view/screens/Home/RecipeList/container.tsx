import React, { FC } from 'react'
import { useObservable } from 'rxjs-hooks'
import UC from 'actions'

import RecipeListComponent from './component'

const RecipeListContainer:FC = () => {
  const { mealsData } = useObservable(UC.HomeScreenService.getSubject)
  || UC.HomeScreenService.getCurrentState()

  return (
    <RecipeListComponent meals={ mealsData } />
  )
}

export default RecipeListContainer
