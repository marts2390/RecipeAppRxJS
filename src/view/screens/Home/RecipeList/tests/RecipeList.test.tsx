import React from 'react'
import { shallow } from 'enzyme'
import RecipeList from '../component'
import mealItemArrayMock from '../../../../../mocks/mealItemArrayMock'

test('should render recipe list', () => {
  const wrapper = shallow(<RecipeList meals={ mealItemArrayMock } />)
  expect(wrapper).toMatchSnapshot()
})
