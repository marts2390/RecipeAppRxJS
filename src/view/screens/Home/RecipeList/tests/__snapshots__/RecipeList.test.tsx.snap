// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render recipe list 1`] = `
<div
  className="recipeList"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="52874"
      md={3}
      sm={4}
      xl={2}
      xs={6}
    >
      <RecipeCardContainer
        item={
          Object {
            "idMeal": "52874",
            "strMeal": "Beef and Mustard Pie",
            "strMealThumb": "https://www.themealdb.com/images/media/meals/sytuqu1511553755.jpg",
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="52878"
      md={3}
      sm={4}
      xl={2}
      xs={6}
    >
      <RecipeCardContainer
        item={
          Object {
            "idMeal": "52878",
            "strMeal": "Beef and Oyster pie",
            "strMealThumb": "https://www.themealdb.com/images/media/meals/wrssvt1511556563.jpg",
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="52997"
      md={3}
      sm={4}
      xl={2}
      xs={6}
    >
      <RecipeCardContainer
        item={
          Object {
            "idMeal": "52997",
            "strMeal": "Beef Banh Mi Bowls with Sriracha Mayo, Carrot & Pickled Cucumber",
            "strMealThumb": "https://www.themealdb.com/images/media/meals/z0ageb1583189517.jpg",
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
